### ISSUE TITLE
fix(metalgemm): evaluate index-ordered Qwen graph norm reduction

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12889]
Parent: #12349
Coordinates with: #12853

## Parent context
#12349 exact Metal MTP parity remains RED. #12853 observed that a host 256-leaf reduction with CPU sqrt64 reciprocal reproduces all Metal norm bits for one captured row; sequential reduction differs.

## Current state
qg_norm reduces 256 lane sums in a tree, while normCfg sums squares in index order. This observed row does not establish universal reciprocal equivalence or sole downstream cause.

## For / Problem / Today / Better because
For native Metal exact-parity qualification, isolate reduction ordering through a minimal shader candidate. Today existing graph normalization differs from serial at the earliest measured operation. Matching the serial order tests that mechanism without changing the oracle.

## Why this is next
The captured norm experiment identifies a falsifiable reduction-order candidate.

## Working spine
Existing qg_norm lane0 accumulates index-increasing squares, computes unchanged rsqrt, shares inverse through threadgroup scalar/barrier, and retains parallel x*inv*gain output.

## Core through-line
One existing kernel arithmetic-order change, exact unchanged parent witness and captured receipt.

## Scope
```routing
lane: metalgemm
paths: ["internal/metalgemm/qwen35_graph.m"]
expected_steps: 3
```

## Likely files
`internal/metalgemm/qwen35_graph.m`

## Scope class
Work unit: leaf
One S1 shader candidate in one file, separate from parent no-shader scope.

## Definition of done
- [ ] Frozen candidate independently reviewed for compiler contraction and arithmetic validity.
- [ ] Granted native device run of unchanged strict witness captures norm and downstream exact results.
- [ ] Review and integration decision retain measured operating limits.

## Gold-plating boundary
No tolerance, reference edits, librarywide fastmath flags, replacement engine or performance promotion. rsqrt and output expression unchanged initially. Serial reduction may regress wide-H latency; measure before performance promotion.

## Acceptance gate
Unchanged TestMetalMTPP4StrictSerialParity must actually execute native Metal. Report norm mismatch/equality and all strict downstream failures separately; no claim that one row qualifies every shape.

## Done condition
Candidate has independently witnessed exact results and an explicit integration disposition; parent closure still requires original complete parity. An arithmetic failure requires a new bounded diagnosis, not relaxed assertions.

## Witness
env -u FAK_METAL_MPS CGO_ENABLED=1 GOMAXPROCS=2 GOFLAGS=-p=1 go test ./internal/model -run '^TestMetalMTPP4StrictSerialParity$' -count=1 -timeout=60s -json

## Budget
Source only until independent review and serialized grant; one physical diagnostic, outer300/test60, unchanged existing fixture workload. No benchmark now.

## Duplicate search
Open/closed norm serial Metal search found diagnostic #12853 and broad prior graph work; none names this index-ordered qg_norm candidate.

## Problem frame
Centrality: Core
Portfolio tier: 2 serving
P1: advanced - exact arithmetic qualification.
P2: preserved - no unmeasured performance claim.
P3: preserved - one kernel and one variable.
P4: advanced - unchanged physical witness.

## Work estimate
Estimate: 1 point.

## Overall completion contribution
Contribution: 1/4 points within #12349; planning only.

## Completion standard
experiment

## Closure binding
Explicit-path witnessed commit cites child and (fak metalgemm); parent stays open for original exact acceptance.

**Comment by anthony-chaudhary** [comment]
Single required-device candidate executed: session8810 exit1, 1 named run/0 skips, fak-native Apple M3 Pro. Index-ordered qg_norm now matches all256 host norm outputs for captured token7 exactly (SHA3248294ecdeabab6b3b89cdd4a104f7c0e07239611b24dbe4a899cfaee1bfa91); A_serial matches and B_tree differs. Unchanged strict witness remains RED for logits/GDN/KV/hidden/continuation; layer0 conv[0] bf7e14f1 versus bf7e14f0. Full source141d7 unchanged before/after. Logs and receipt in managedmetalgemm559b86d76146/_scratch/issue-12889-hil/. CPU/HIL slot and lease released, no retry. This fixes only the measured norm boundary; no universal arithmetic, wide-H latency, performance or parent closure claim. Next: independently review remaining first projection boundary and candidate operating envelope.
